probe/endpoint/dns_snooper_linux_amd64.go

```diff
@@ -29,6 +30,14 @@ type DNSSnooper struct {
 	decodingErrorCounts map[string]uint64 // for limiting
 }
 
+type domainSet struct {
+	// Not worth using a RMutex since we don't expect a lot of contention
+	// and they are considerably larger (8 bytes vs 24 bytes), which would
+	// bloat the cache
+	sync.Mutex
```

Comment:
It's not gcache that you need to worry about but the map entries. Which certainly *do* hold structs, not pointers.